neuro_extras/data/remote.py

```diff
@@ -0,0 +1,230 @@
+"""Module for copying files by running neu.ro jobs"""
+import logging
+from dataclasses import dataclass
+from typing import List, Mapping, Optional, Tuple
+
+from neuro_sdk import Client, DiskVolume, RemoteImage, SecretFile, Volume
+from yarl import URL
+
+from ..common import EX_OK, NEURO_EXTRAS_IMAGE, _attach_job_stdout
+from .common import (
+    Copier,
+    UrlType,
+    get_default_preset,
+    get_filename_from_url,
+    parse_resource_spec,
+    strip_filename_from_url,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class RemoteJobConfig:
+    """Arguments, passed to `neuro_sdk.Client.jobs.start()`"""
+
+    image: RemoteImage
+    command: str
+    env: Optional[Mapping[str, str]]
+    secret_env: Optional[Mapping[str, URL]]
+    volumes: List[Volume]
+    secret_files: List[SecretFile]
+    disk_volumes: List[DiskVolume]
+    preset_name: str
+    life_span: Optional[float]
+    pass_config: bool
+
+    @staticmethod
+    def create(
+        source: str,
+        destination: str,
+        neuro_client: Client,
+        compress: bool = False,
+        extract: bool = False,
+        volumes: Optional[List[str]] = None,
+        env: Optional[List[str]] = None,
+        preset: Optional[str] = None,
+        life_span: Optional[float] = None,
+    ) -> "RemoteJobConfig":
+        """Create `RemoteJobConfig` for a neu.ro copy job.
+
+        Copy job will copy data from `source` to `destination`"""
+        image = neuro_client.parse.remote_image(NEURO_EXTRAS_IMAGE)
+
+        (patched_source, patched_destination, data_mounts,) = _map_into_volumes(
+            source=source,
+            destination=destination,
+        )
+        command = _build_data_copy_command(
+            source=patched_source,
+            destination=patched_destination,
+            extract=extract,
+            compress=compress,
+        )
+        all_volumes = volumes + data_mounts if volumes else data_mounts
+        env_parse_result = neuro_client.parse.envs(env if env else [])
+        volume_parse_result = neuro_client.parse.volumes(all_volumes)
+        preset_name = preset or get_default_preset(neuro_client)
```

Comment:
Seems like `select_build_preset` does not guarantee it can return a valid preset, which would still require a call to get_default_preset in something like 
```python
preset_name = preset or select_build_preset(...) or get_default_preset(...)
```